cmd/ova-parser/ova-parser.go

```diff
@@ -0,0 +1,336 @@
+	OvaPath               string
+	RevisionValidated     int64
+	PolicyVersion         int
+	UUID                  string
+	Firmware              string
+	CpuAffinity           []int32
+	CpuHotAddEnabled      bool
+	CpuHotRemoveEnabled   bool
+	MemoryHotAddEnabled   bool
+	FaultToleranceEnabled bool
+	CpuCount              int32
+	CoresPerSocket        int32
+	MemoryMB              int32
+	BalloonedMemory       int32
+	IpAddress             []string
+	NumaNodeAffinity      []string
+	StorageUsed           int64
+	ChangeTrackingEnabled bool
+	Devices               []Device
+	NICs                  []NIC
+	Disks                 []VmDisk
+	Networks              []Network
+}
+
+// Virtual Disk.
+type VmDisk struct {
+	FilePath                string
+	Capacity                string
+	CapacityAllocationUnits string
+	DiskId                  string
+	FileRef                 string
+	Format                  string
+	PopulatedSize           string
+}
+
+// Virtual Device.
+type Device struct {
+	Kind string `json:"kind"`
+}
+
+type Conf struct {
+	key   string
+	Value string
+}
+
+// Virtual ethernet card.
+type NIC struct {
+	Name   string `json:"name"`
+	MAC    string `json:"mac"`
+	Config []Conf
+}
+
+func main() {
+	ovaFiles, err := findOVAFiles("/ova")
+	if err != nil {
+		fmt.Println("Error finding OVA files:", err)
+		return
+	}
+
+	for _, ovaFile := range ovaFiles {
+		fmt.Println("Processing OVA file:", ovaFile)
+
+		ovfPath, tmpDir, err := extractOVFFromOVA(ovaFile)
+		if err != nil {
+			fmt.Println("Error extracting OVF from OVA:", err)
+			continue
+		}
+
+		vmXML, err := processOVF(ovfPath)
+		if err != nil {
+			fmt.Println("Error processing OVF file:", err)
+			os.RemoveAll(tmpDir)
+		}
+
+		vmStruct, err := convertToVmStruct(*vmXML, ovfPath)
+		if err != nil {
+			fmt.Println("Error processing OVF file:", err)
+			os.RemoveAll(tmpDir)
+		}
+
+		fmt.Println(vmStruct)
+		os.RemoveAll(tmpDir)
+	}
+}
+
+func findOVAFiles(directory string) ([]string, error) {
+	var ovaFiles []string
+
+	err := filepath.Walk(directory, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !info.IsDir() && strings.HasSuffix(strings.ToLower(info.Name()), ".ova") {
+			ovaFiles = append(ovaFiles, path)
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return ovaFiles, nil
+}
+
+func extractOVFFromOVA(ovaFile string) (string, string, error) {
+	tmpDir, err := ioutil.TempDir("", "ova_extraction")
+	print(tmpDir)
+	if err != nil {
+		os.RemoveAll(tmpDir)
+		return "", "", err
+	}
+
+	cmd := exec.Command("tar", "-xf", ovaFile, "-C", tmpDir, "*.ovf")
+	err = cmd.Run()
+	if err != nil {
+		os.RemoveAll(tmpDir)
+		return "", "", err
+	}
+
+	ovfFiles, err := filepath.Glob(filepath.Join(tmpDir, "*.ovf"))
+	if err != nil {
+		os.RemoveAll(tmpDir)
+		return "", "", err
+	}
+
+	if len(ovfFiles) == 0 {
+		os.RemoveAll(tmpDir)
+		return "", "", fmt.Errorf("no OVF file found in the OVA")
+	}
```

Comment:
Since you only use first OVF you may want to log a warning if there are more files in the archive.